### PR TITLE
websocket integration

### PR DESCRIPTION
**Pull request by shorsher**
- auto refresh views when new events are received
 - update transaction status types
 - update view & interfaces to match latest API spec

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,6 +23,7 @@
     "react-i18next": "^11.8.15",
     "react-router-dom": "^5.2.0",
     "react-scripts": "4.0.3",
+    "reconnecting-websocket": "^4.4.0",
```
Comment (awrichar) [inline package.json:26]: ❤️

File: src/index.tsx [hunk 2]
```diff
@@ -32,9 +32,4 @@ i18n.use(initReactI18next).init({
   },
 });
 
-ReactDOM.render(
-  <React.StrictMode>
-    <App />
-  </React.StrictMode>,
-  document.getElementById('root')
-);
+ReactDOM.render(<App />, document.getElementById('root'));
```
Comment (awrichar) [inline src/index.tsx:35]: No more StrictMode?

**Review by awrichar — APPROVED** [review]
Looks generally good - left a few questions inline but doesn't need to hold anything up.

File: src/App.tsx [hunk 3]
```diff
@@ -116,6 +120,20 @@ function App() {
       });
   }, []);
 
+  useEffect(() => {
+    if (selectedNamespace) {
+      ws.current = new ReconnectingWebsocket(
+        `${protocol}://${window.location.hostname}:5000/ws?namespace=${selectedNamespace}&ephemeral&autoack`
```
[the remark this answers is not in the document]
Comment (shorsher) [reply src/App.tsx]: good catch 🙏

File: src/index.tsx (hunk 2 above)
Comment (shorsher) [reply src/index.tsx:35]: 👀

Comment (shorsher) [reply src/index.tsx]: fixed, thanks

**Pull request merged**